node/src/actor_system_factory.rs

```diff
@@ -1866,37 +1877,86 @@ mod tests {
                 _ => low(high(previous)),
             }
         }
+        fn assert_is_not_trait_definition(
+            body_lines: impl Iterator<Item = String>,
+        ) -> impl Iterator<Item = String> {
+            fn yield_if_contains_semicolon(line: &str) -> Option<String> {
+                line.contains(';').then(|| line.to_string())
+            }
+            let mut semicolon_line_opt = None;
+            let vec_lines = body_lines
+                .map(|line| {
+                    if semicolon_line_opt.is_none() {
+                        if let Some(result) = yield_if_contains_semicolon(&line) {
+                            semicolon_line_opt = Some(result)
+                        }
+                    }
+                    line
+                })
+                .collect::<Vec<_>>();
+            if let Some(line) = semicolon_line_opt {
+                let regex = Regex::new(r"Vec<\w+>;").unwrap();
+                if regex.is_match(&line) {
+                    //means the function body is empty -> trait definition
+                    panic!("the second parsed chunk of code is a trait definition and the implementation lies first")
+                }
+            } else {
+                () //means is a clean function body without semicolon
+            }
+            vec_lines.into_iter()
+        }
 
-        let file_path = current_dir()
-            .unwrap()
+        let current_dir = current_dir().unwrap();
+        let file_path = current_dir
             .join("src")
             .join("accountant")
             .join("receivable_dao.rs");
-        let file = File::open(file_path).unwrap();
+        let file = match File::open(file_path) {
+            Ok(file) => file,
+            Err(_) => {
+                if !is_running_under_github_actions() && !is_source_code_attached(&current_dir) {
+                    return Skip;
+                } else {
+                    panic!(
```

Comment:
Suggestion: "This test requires access to source code to run, but cannot find source code in this environment. If that does not surprise you, set the environment variable MASQ_SOURCE_CODE_UNAVAILABLE to some non-blank value and run the tests again."